frappe/client.py

```diff
@@ -312,3 +313,37 @@ def get_js(items):
 def get_time_zone():
 	'''Returns default time zone'''
 	return {"time_zone": frappe.defaults.get_defaults().get("time_zone")}
+
+@frappe.whitelist()
+def attach_file():
+	'''Attach a file to Document (POST)
+
+	:param filename: filename e.g. test-file.txt
+	:param filedata: base64 encode filedata
+	:param doctype: Reference DocType to attach file
+	:param docname: Reference DocName to attach file
+	:param folder: Folder to add File into
+	:param decode: decode filedata from base64 encode
+	:param is_private: Attach file as private file
+	:param docfield: file to attach to'''
+
+	filename = frappe.form_dict.get("filename")
```

Comment:
pass as function params?

Reply:
M using frappe.form_dict.get() because it needs to be body params of POST request.

Params to the function will result in url queries of GET request.